qtfred/src/CheckedDataListModel.h

```diff
@@ -141,21 +141,84 @@ class CheckedDataListModel : public QAbstractListModel
 			return defaultFlags;
 	}
 
-	inline const SCP_unordered_set<T*>& getCheckedData() {
-		return checkeds;
+private:
+	template<typename V>
+	inline void getCheckedDataInternal(SCP_unordered_set<V*> &set) {
+		for (auto& item : items)
+			if (item._checked)
+				set.insert(&item.internalData());
 	}
 
-	inline const SCP_unordered_set<const T*>& getCheckedDataConst() const {
-		return checkedsConst;
+public:
+	inline SCP_unordered_set<T*> getCheckedData() {
+		SCP_unordered_set<T*> ret{items.size() / 2};
+		getCheckedDataInternal(ret);
+		return ret;
 	}
 
+	inline SCP_unordered_set<const T*> getCheckedData() const {
```

Comment:
In contexts where only a const reference/pointer to the model is available (or necessary), only read access to the managed data can be given. In other contexts it may be necessary to find and edit one or more specific checked items.